docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,130 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+OpenShift cluster admins need a way to sync their OpenShift `Groups` to external records in an LDAP server. This requires that we have a way of retrieving the LDAP users that are the members of a group, determining the OpenShift `Group` name for this collection, and updating the OpenShift record, removing the record as necessary when the corresponding LDAP record is removed. Necessary information about the LDAP server set-up and schema will be provided by the user in a `yaml` configuration file and referenced in the `oadm sync-groups` command.
+
+## Use Cases
+1. I want to sync all LDAP groups minus some blacklist to OpenShift `Groups`
+* I want to sync a specific set of LDAP groups to OpenShift `Groups`
+* I want to sync all OpenShift groups minus some blacklist with the LDAP server
```

Comment:
Also, we need to support that type of mapping because we're also supporting the user specifying how to name an OpenShift `Group` given the LDAP group entry data.